std/haxe/math/bigint/BigIntExceptions.hx

```diff
@@ -19,17 +19,16 @@
  * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
  * DEALINGS IN THE SOFTWARE.
  */
- 
+
 package haxe.math.bigint;
 
 /* Original code courtesy Chuck Batson (github.com/cbatson) */
-class BigIntExceptions
-{
-    public static inline var INVALID_ARGUMENT = "Invalid argument";
-    public static inline var BUFFER_TOO_SMALL = "Buffer too small";
-    public static inline var DIVISION_BY_ZERO = "Division by zero";
-    public static inline var NEGATIVE_EXPONENT = "Negative exponent";
-    public static inline var INVALID_OPERATION = "Invalid operation";
-    public static inline var NEGATIVE_MODULUS = "Modulus should be positive";
-    public static inline var EVEN_VALUES = "Both values are even";
-}
\ No newline at end of file
+enum abstract BigIntExceptions(String) {
```

Comment:
The tests are failing probably because this need to be changed to 
```haxe
enum abstract BigIntExceptions(String) to String {
```

Reply:
What should be correct way to implement exception here ?
1) Extend [haxe.Exception](https://api.haxe.org/haxe/Exception.html) class .
2) Add ``` to String ``` in ```enum abstract```
3) Change ``eq`` with ```assertThrowsString``` ( similiiar to [hxbitcoin test case](https://github.com/cbatson/hxBitcoin/blob/0c4f9845c33dd26d3fd5f931a0f364931ca7d786/com/fundoware/engine/test/FunTestCase.hx#L51) )

Reply:
@Simn is there any consensus how new types of exception are added to the SDK? Is throwing strings (still) best practices or should custom classes which extend haxe.Exception be thrown?